src/test/spec/index_management.rs

```diff
@@ -272,3 +274,117 @@ async fn search_index_drop_not_found() {
+        return;
+    }
+    let client = Client::test_builder().build().await;
+    let coll_name = ObjectId::new().to_hex();
+    let db = client.database("search_index_test");
+    db.create_collection(&coll_name).await.unwrap();
+    let coll0 = db.collection::<Document>(&coll_name);
+
+    let name = coll0
+        .create_search_index(
+            SearchIndexModel::builder()
+                .name(String::from("test-search-index-case7-implicit"))
+                .definition(doc! { "mappings": { "dynamic": false } })
+                .build(),
+        )
+        .await
+        .unwrap();
+    assert_eq!(name, "test-search-index-case7-implicit");
+    let index1 = wait_for_index(&coll0, &name).await;
+    assert_eq!(index1.get_str("type"), Ok("search"));
+
+    let name = coll0
+        .create_search_index(
+            SearchIndexModel::builder()
+                .name(String::from("test-search-index-case7-explicit"))
+                .index_type(SearchIndexType::Search)
+                .definition(doc! { "mappings": { "dynamic": false } })
+                .build(),
+        )
+        .await
+        .unwrap();
+    assert_eq!(name, "test-search-index-case7-explicit");
+    let index2 = wait_for_index(&coll0, &name).await;
+    assert_eq!(index2.get_str("type"), Ok("search"));
+
+    let name = coll0
+        .create_search_index(
+            SearchIndexModel::builder()
+                .name(String::from("test-search-index-case7-vector"))
+                .index_type(SearchIndexType::VectorSearch)
+                .definition(doc! {
+                    "fields": [{
+                        "type": "vector",
+                        "path": "plot_embedding",
+                        "numDimensions": 1536,
+                        "similarity": "euclidean",
+                    }]
+                })
+                .build(),
+        )
+        .await
+        .unwrap();
+    assert_eq!(name, "test-search-index-case7-vector");
+    let index3 = wait_for_index(&coll0, &name).await;
+    assert_eq!(index3.get_str("type"), Ok("vectorSearch"));
+}
+
+// SearchIndex Case 8: Driver requires explicit type to create a vector search index
+#[tokio::test]
+async fn search_index_requires_explicit_vector() {
+    if env::var("INDEX_MANAGEMENT_TEST_PROSE").is_err() {
+        log_uncaptured("Skipping index management prose test: INDEX_MANAGEMENT_TEST_PROSE not set");
+        return;
+    }
+    let client = Client::test_builder().build().await;
+    let coll_name = ObjectId::new().to_hex();
+    let db = client.database("search_index_test");
+    db.create_collection(&coll_name).await.unwrap();
+    let coll0 = db.collection::<Document>(&coll_name);
+
+    let result = coll0
+        .create_search_index(
+            SearchIndexModel::builder()
+                .name(String::from("test-search-index-case7-vector"))
+                .index_type(SearchIndexType::VectorSearch)
```

Comment:
The [definition](https://github.com/mongodb/specifications/blob/master/source/index-management/tests/README.md#case-8-driver-requires-explicit-type-to-create-a-vector-search-index) for this test doesn't include the `type` field in the index model -- this probably shouldn't be passing written as-is? I think the `create_search_index` call here is equivalent to the last one in the test above

Reply:
As it turns out, the test _was_ failing (i.e. the index creation was succeeding), but when I kicked off the evergreen run I accidentally left in a restriction that filtered out the test entirely.  I've fixed the test and also updated the evergreen config to include the test-level breakdown which would have made that more obvious.  New passing run is [here](https://spruce.mongodb.com/task/mongo_rust_driver_search_index_test_search_index_patch_45736f241311cf2e2d0a44708e421c8cd9282203_667d96433aed0b0007e32def_24_06_27_16_41_44?execution=0&sortBy=STATUS&sortDir=ASC) :)